packages/cli/src/executionLifecycleHooks/hasBinaryData.ts

```diff
@@ -0,0 +1,11 @@
+import type { IRun } from 'n8n-workflow';
+
+export function hasBinaryData(run: IRun) {
+	const { runData } = run.data.resultData;
+
+	return Object.keys(runData).some((nodeName) => {
```

Comment:
My concern here is that we're looking at the first `runIndex` and only the `firstItem` of a node, and maybe we should iterate over all nodes and all run indices and check if _any_ contains binary data right?

Maybe a more assertive approach is to use the `NodeExecuteFunctions` helper functions and flag it whenever we "create" a binary file and flag it as part of the `packages/workflow/src/Workflow.ts` instance.

Reply:
Good point, but to clarify this needs to be part of the execution, not the workflow.